### PR TITLE
add test for num of columns

### PR DESCRIPTION
**Pull request by Robert-Costello**
## Technical Summary

This add a tests to keep the number of columns in case tile templates in parity with what mobile can handle, currently 12.
jira ticket: https://dimagi-dev.atlassian.net/browse/USH-3332

## Feature Flag

relates to Case Tiles feature flags: https://confluence.dimagi.com/display/saas/Allow+Configuration+of+Case+List+Tiles
### Automated test coverage

this is a test

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Robert-Costello** [comment]
After chatting with @nospame and looking more closely at the related android code, I realized my approaching to this test was incorrect. Converting to draft until it's ready for review.

**Comment by Robert-Costello** [comment]
The updated test checks that the farthest reaching field in a case tile template does not extend past the limit, currently 12. If two fields share a y coordinate (sit side-by-side), as long as the starting x coord plus the width of the right-most field are within the bounds the test will pass.

File: corehq/apps/app_manager/tests/test_suite_case_tiles.py [hunk 1]
```diff
@@ -114,6 +114,19 @@ def add_columns_for_one_one_two_case_details(_module):
...
     file_path = ('data', 'suite')
 
+    # Keeps the number of columns in parity with what mobile allows
+    def test_case_tile_column_count(self):
+        for choice in CaseTileTemplates.choices:
+            template_name = choice[0]
+            template_grid = case_tile_template_config(template_name).grid
+
+            for field in template_grid.values():
+                absWidth = field.get('x') + field.get('width')
```
Comment (nospame) [inline corehq/apps/app_manager/tests/test_suite_case_tiles.py:124]: I think this should be `absolute_width` unless there's a clear reason for `absWidth`

Comment (Robert-Costello) [reply corehq/apps/app_manager/tests/test_suite_case_tiles.py:124]: Agreed, no reason, just accidental camel case.

**Comment by nospame** [comment]
> The updated test checks that the farthest reaching field in a case tile template does not extend past the limit, currently 12. If two fields share a y coordinate (sit side-by-side), as long as the starting x coord plus the width of the right-most field are within the bounds the test will pass.

Looking at ticket comments, is this approach in line with "calculate the total width for fields sharing a y coord"? I'm not sure what the real-life behavior is if two fields overlap -- maybe doesn't actually matter whether they're calculated individually or as a row.

**Comment by Robert-Costello** [comment]
> I'm not sure what the real-life behavior is if two fields overlap -- maybe doesn't actually matter whether they're calculated individually or as a row.

In terms of maintaining the maximum width, I think this approach does what we want. Functionally it accounts for the full width of fields that share a y coord. My initial approach was a more complicated algo that mapped fields by y coordinate in a dict and calculated the absolute width of each dict entry. The math for that approach is simplified down to what's in this PR, but it doesn't account for other template formatting concerns like overlap. I think we could address things like overlap in additional tests if needed.

**Review by nospame — APPROVED** [review]

**Pull request merged**